core/session.ts

```diff
@@ -515,6 +518,12 @@ export class Session {
       action.dependencies = (action.dependencyTargets || []).map(dependencyTarget =>
         utils.targetToName(dependencyTarget)
       );
+
+      if ("parent" in action) {
+        if (!!action.parent) {
+          action.parent = newTargetByOriginalTarget.get(action.parent);
+        }
+      }
```

Comment:
Any idea why `hasOwnProperty` doesn't work here?

Reply:
I'd presume because the object prototype is where `parent` is defined, and thus the object inherits it (and so `hasOwnProperty` will return false for `parent`).